templates/commands/render/render_action_include_test.go

```diff
@@ -58,6 +58,67 @@ func TestActionInclude(t *testing.T) {
 				"myfile.txt": {0o600, "my file contents"},
 			},
 		},
+		{
+			name: "file_name_wildcard_globbing",
+			include: &spec.Include{
+				Paths: []*spec.IncludePath{
+					{
+						Paths: modelStrings([]string{"hello?.txt"}),
+					},
+				},
+			},
+			templateContents: map[string]modeAndContents{
+				"hello.txt":  {0o600, "hello file contents"},
+				"hello1.txt": {0o600, "hello 1 file contents"},
+				"hello2.txt": {0o600, "hello 2 file contents"},
+			},
+			wantScratchContents: map[string]modeAndContents{
+				"hello1.txt": {0o600, "hello 1 file contents"},
+				"hello2.txt": {0o600, "hello 2 file contents"},
+			},
+		},
+		{
+			name: "file_name_star_globbing",
+			include: &spec.Include{
+				Paths: []*spec.IncludePath{
+					{
+						Paths: modelStrings([]string{"hello*.txt"}),
+					},
+				},
+			},
+			templateContents: map[string]modeAndContents{
+				"hello_its_me.txt":       {0o600, "hello its me file contents"},
+				"hellooo.txt":            {0o600, "hellooo file contents"},
+				"hello-test.txt":         {0o600, "hello-test file contents"},
+				"hey.txt":                {0o600, "hey file contents"},
+				"dont-include-hello.txt": {0o600, "dont include hello file contents"},
+			},
+			wantScratchContents: map[string]modeAndContents{
+				"hello_its_me.txt": {0o600, "hello its me file contents"},
+				"hellooo.txt":      {0o600, "hellooo file contents"},
+				"hello-test.txt":   {0o600, "hello-test file contents"},
+			},
+		},
+		{
+			name: "file_extension_globbing",
```

Comment:
Maybe another test case where the glob matches a directory name? Like `a/*/b.txt`

Reply:
It's also probably worth noting that `a/*/b.txt` will include `a/foo/b.txt` and `a/bar/b.txt`, but **not** `a/foo/bar/b.txt` since `*` matches any sequence of **non-Separator** characters

Reply:
Test case star_glob_paths covers directory matching.